include/soci/logger.h

```diff
@@ -38,6 +43,7 @@ class SOCI_DECL logger_impl
     virtual void set_stream(std::ostream * s);
     virtual std::ostream * get_stream() const;
     virtual std::string get_last_query() const;
+    virtual std::string get_last_query_with_context() const;
```

Comment:
I'd call this `get_last_query_context()` and return the string with the parameter values from it (and empty string by default) and append it to the query in `get_last_query()` if it's non-empty.

This doesn't really change much, of course, but it just looks more logical to have a function doing A and a function doing B, with A using B, instead of having a function doing A and a function doing A+B.

Reply:
Sounds good